### PR TITLE
[Feature] development Show Load ALL SQL 

### PR DESCRIPTION
**Pull request by qimumu9406**
# Proposed changes

Issue Number:  #9475

## Problem Summary:
show load all

Describe the overview of changes.
user role is root and admin action show load all

## Checklist(Required)

1. Does it affect the original behavior: Yes
2. Has unit tests been added: Yes
3. Has document been added or modified: Yes
4. Does it need to update dependencies: No
5. Are there any changes that cannot be rolled back: No

## Further comments
Filter out mysql schema database by db id and db name

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/zh-CN/sql-manual/sql-reference/Show-Statements/SHOW-LOAD.md [hunk 1]
```diff
@@ -51,20 +51,22 @@ SHOW LOAD
 说明：
 
 1) 如果不指定 db_name，使用当前默认db
+
+1) db_name 和 ALL 只能选择一个，当使用 ALL 时(目前支持root用户使用)，将查询集群所有db的导入任务
     
-2)  如果使用 LABEL LIKE，则会匹配导入任务的 label 包含 label_matcher 的导入任务
+1)  如果使用 LABEL LIKE，则会匹配导入任务的 label 包含 label_matcher 的导入任务
```
Comment (stalary) [inline docs/zh-CN/sql-manual/sql-reference/Show-Statements/SHOW-LOAD.md:57]: There seems to be something wrong with the number?

File: docs/zh-CN/sql-manual/sql-reference/Show-Statements/SHOW-LOAD.md [hunk 2]
```diff
@@ -78,40 +80,46 @@ SHOW LOAD
...
     ```
 
-2. 展示指定 db 的导入任务，label 中包含字符串 "2014_01_02"，展示最老的10个
+1. 展示所有db的所有导入任务(root 用户执行)
+    
+   ```sql
+    SHOW LOAD ALL;
+    ```
+
+1. 展示指定 db 的导入任务，label 中包含字符串 "2014_01_02"，展示最老的10个
```
Comment (stalary) [inline docs/zh-CN/sql-manual/sql-reference/Show-Statements/SHOW-LOAD.md:89]: There seems to be something wrong with the number?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowLoadStmt.java [hunk 3]
```diff
@@ -61,6 +62,17 @@ public class ShowLoadStmt extends ShowStmt {
...
         this.whereClause = labelExpr;
+        this.isAll=false;
+        this.orderByElements = orderByElements;
+        this.limitElement = limitElement;
+
+        this.labelValue = null;
+        this.stateValue = null;
+        this.isAccurateMatch = false;
+    }
+    public ShowLoadStmt(boolean isAll, Expr labelExpr, List<OrderByElement> orderByElements, LimitElement limitElement) {
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowLoadStmt.java:73]: Add a newline

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowLoadStmt.java [hunk 4]
```diff
@@ -61,6 +62,17 @@ public class ShowLoadStmt extends ShowStmt {
     public ShowLoadStmt(String db, Expr labelExpr, List<OrderByElement> orderByElements, LimitElement limitElement) {
         this.dbName = db;
         this.whereClause = labelExpr;
+        this.isAll=false;
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowLoadStmt.java:65]: There are blank before and after the equals sign.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowLoadStmt.java [hunk 5]
```diff
@@ -256,13 +273,16 @@ private void analyzeSubPredicate(Expr subExpr) throws AnalysisException {
     @Override
     public String toSql() {
         StringBuilder sb = new StringBuilder();
-        sb.append("SHOW LOAD ");
-        if (!Strings.isNullOrEmpty(dbName)) {
-            sb.append("FROM `").append(dbName).append("`");
+        if(isAll){
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowLoadStmt.java:276]: format.

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 6]
```diff
@@ -993,21 +995,40 @@ private void handleShowLoad() throws AnalysisException {
         ShowLoadStmt showStmt = (ShowLoadStmt) stmt;
 
         Catalog catalog = Catalog.getCurrentCatalog();
-        Database db = catalog.getDbOrAnalysisException(showStmt.getDbName());
-        long dbId = db.getId();
+        List<Database> dbList=Lists.newArrayList();
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:998]: format.
http://doris.incubator.apache.org/zh-CN/developer-guide/java-format-code.html

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 7]
```diff
@@ -993,21 +995,40 @@ private void handleShowLoad() throws AnalysisException {
...
+
+        if(showStmt.getIsAll()){
+            if(ctx.getCurrentUserIdentity().getQualifiedUser().equals("root")){
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:1001]: suggest
```java
ctx.getCurrentUserIdentity().isRootUser()
```

File: fe/fe-core/src/test/java/org/apache/doris/analysis/ShowLoadStmtTest.java [hunk 8]
```diff
@@ -78,6 +80,13 @@ public void testNormal() throws UserException, AnalysisException {
         Assert.assertEquals("SHOW LOAD FROM `testCluster:testDb`", stmt.toString());
     }
 
+    @Test
+    public void testAllDb() throws UserException, AnalysisException {
```
Comment (stalary) [inline fe/fe-core/src/test/java/org/apache/doris/analysis/ShowLoadStmtTest.java:84]: You can replace the old tests with extends TestWithFeService

File: fe/fe-core/src/test/java/org/apache/doris/analysis/ShowLoadStmtTest.java [hunk 9]
```diff
@@ -122,6 +131,31 @@ public void testWhere() throws UserException, AnalysisException {
         stmt.analyze(analyzer);
         Assert.assertEquals("SHOW LOAD FROM `testCluster:testDb` WHERE `state` = \'PENDING\' LIMIT 10", stmt.toString());
     }
+    @Test
+    public void testAllDbWhere() throws UserException, AnalysisException {
+        ShowLoadStmt stmt = new ShowLoadStmt(true, null, null, null);
+        stmt.analyze(analyzer);
+        Assert.assertEquals("SHOW LOAD ALL ", stmt.toString());
+
+        SlotRef slotRef = new SlotRef(null, "label");
```
Comment (stalary) [inline fe/fe-core/src/test/java/org/apache/doris/analysis/ShowLoadStmtTest.java:140]: use parseAndAnalyzeStmt.

File: fe/fe-core/src/test/java/org/apache/doris/analysis/ShowLoadStmtTest.java [hunk 10]
```diff
@@ -156,5 +190,39 @@ public void testInvalidWhereClause() throws AnalysisException {
                 () -> stmt2.analyze(analyzer));
 
 
+    }
+    @Test
+    public void testAllDbInvalidWhereClause() throws AnalysisException {
+        //test:  WHERE label="abc" AND label LIKE "def";  --> AnalysisException
+        SlotRef slotRef1 = new SlotRef(null, "label");
```
Comment (stalary) [inline fe/fe-core/src/test/java/org/apache/doris/analysis/ShowLoadStmtTest.java:197]: use parseAndAnalyzeStmt.

**Comment by caiconghui** [comment]
actually, there are many similar overview for the whole cluster, it is not a good design to show all load for all databases at one sql, how about like show proc "/transactions", we can execute show proc "/load_jobs", then we can see the summary of all the database load, and then we can see load detail for any database when we choose one database id, to keep the same style for the doris.

**Comment by stalary** [comment]
Show All Load and show All Routineload have different semantics. We really should redesign this feature.

**Comment by qimumu9406** [comment]
Thanks for your suggestions, I think your suggestions are very helpful to me. Show Load ALL should be to get the ALL Load that the current user has DB permissions.

**Pull request closed without merging**